examples/apiserver/apiserver_test.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apiserver
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"testing"
+	"time"
+
+	"k8s.io/kubernetes/cmd/libs/go2idl/client-gen/testdata/apis/testgroup/v1"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+)
+
+var serverIP = "http://localhost:8080"
```

Comment:
Is it possible to use another port? If I have a local cluster running via hack/local-cluster-up.sh, it would conflict with this test.

The log actually says `I0204 13:52:20.157799    9217 genericapiserver.go:607] Serving securely on 0.0.0.0:6443`, I don't have time to dig into the code but it seems the generic apiserver randoms picks a port if 8080 is not available. If that's the case, we probably should improve this test to be more robust.

Sorry I didn't catch these yesterday.